ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanProperty.java

```diff
@@ -759,12 +765,17 @@ public Object value(Object bean) {
     return getValueIntercept((EntityBean) bean);
   }
 
+  private boolean beanHasProperty(EntityBean bean) {
+    return descriptor.inheritInfo() == null // fast exit for non inherited beans
+      || owningType.isInstance(bean);
```

Comment:
PLEASE CHECK: 
the idea is, not to call `isInstance` when no inheritance is used (90% of use cases)
BUT there is no barrier somewhere that prevents the user to use a property from model A on model B.

So the question is, should we pay an additional price for the instance check here to make ebean more robust.

Reply:
This seems reasonable.

Q: The only way to use an incorrect property is via `io.ebean.plugin.Property` correct?

Reply:
We sometimes use ebeaninternal.BeanProperty :-) but there is also ExpressionPath.pathGet/Set in the public API